src/components/Upload.tsx

```diff
@@ -64,11 +68,77 @@ export const Upload = () => {
   // display file names
   const [files, setFiles] = useState<string[]>([])
 
+  // two buttons
+  const [showIndividual, setShowIndividual] = useState<boolean>(false)
+
+  const [showFull, setShowFull] = useState<boolean>(false)
+  let stringSplitExcel = JSON.stringify(splitExcel)
   // // checking aggregate
+  const [testing, setTesting] = useState<any>({})
+
   useEffect(() => {
+    console.log(testing, 'this is testing')
+    if (Object.keys(splitExcel).length > 0) {
+      splittingFunction(splitExcel)
+
+      setAggregate(() => testing)
+
+    }
 
-  }, [aggregate, splitExcel, files.length])
+  }, [JSON.stringify(testing), JSON.stringify(aggregate), stringAgg, Object.keys(splitExcel).length, files.length, showFull, showIndividual])
+
+  // parses aggregate information
+  const splittingFunction = async (splitExcel) => {
+
+    const filesNames = Object.keys(splitExcel)
+
+    filesNames.forEach((file, index) => {
+      let fileInfo = splitExcel[file]['unitInfo'];
+      let [year, month, buildingName] =  [fileInfo[0]['Año'], fileInfo[0]['Mes'].toLowerCase(), fileInfo[0]['Depto']]
```

Comment:
is this correct? BuildingName variable is taking 'Depto' (Apartment) variable from file.